src/openforms/contrib/kadaster/api/tests/test_api_map_search.py

```diff
@@ -0,0 +1,248 @@
+from unittest.mock import MagicMock, patch
+
+from django.test import override_settings
+
+import requests
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APITestCase
+
+from openforms.submissions.tests.factories import SubmissionFactory
+from openforms.submissions.tests.mixins import SubmissionsMixin
+
+
+@override_settings(LANGUAGE_CODE="en")
+class MapSearchApiTests(SubmissionsMixin, APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.submission = SubmissionFactory.create()
+
+    def tearDown(self):
+        self._clear_session()
+
+    @patch("openforms.contrib.kadaster.api.views.requests")
+    def test_call_with_query_parameter_utrecht(self, mock_requests):
+        # mock the response
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {
+            "response": {
+                "numFound": 205959,
+                "start": 0,
+                "maxScore": 9.7794895,
+                "numFoundExact": True,
+                "docs": [
+                    {
+                        "bron": "Bestuurlijke Grenzen",
+                        "identificatie": "0344",
+                        "provinciecode": "PV26",
+                        "type": "gemeente",
+                        "provincienaam": "Utrecht",
+                        "centroide_ll": "POINT(5.0747543 52.09113798)",
+                        "gemeentecode": "0344",
+                        "weergavenaam": "Gemeente Utrecht",
+                        "provincieafkorting": "UT",
+                        "centroide_rd": "POINT(133587.182 455921.594)",
+                        "id": "gem-df0ca8ab37eccea5217e2a13f74d2833",
+                        "gemeentenaam": "Utrecht",
+                        "score": 9.7794895,
+                    },
+                ],
+            }
+        }
+
+        mock_requests.get.return_value = mock_response
+        url = reverse("api:geo:address-search")
+
+        self._add_submission_to_session(self.submission)
+
+        response = self.client.get(url, {"q": "utrecht"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        body = response.json()
+        self.assertEqual(
+            body,
+            [
+                {
+                    "label": "Gemeente Utrecht",
+                    "latLng": {"lat": 52.09113798, "lng": 5.0747543},
+                    "rd": {"x": 133587.182, "y": 455921.594},
+                },
+            ],
+        )
```

Comment:
can you please break up your test itself in blocks my removing whitespace for the setup and assert phase and ensure there's a blank line between each phase?

```
# setup phase
setup1
setup2

# call phase
call_the_thing_you_test

# assert phase
assert1
assert2
```

Sticking to this structure helps in reading tests and figuring out what's actually being tested.